test/test_transforms_v2_refactored.py

```diff
@@ -312,11 +312,12 @@ def adapt_fill(value, *, dtype):
         return value
 
     max_value = get_max_value(dtype)
+    value_type = float if dtype.is_floating_point else int
```

Comment:
If we adapt the fill value, we also want to adapt the type. I've checked and this does not hide the errors that we are fixing here.